keps/sig-apps/20190318-Graduate-CronJob-to-Stable.md

```diff
@@ -0,0 +1,575 @@
+
+```golang
+delta = nextScheduleTime - currentTime
+jitter = delta*cronjob.Spec.Jitter/100
+nextScheduleTime += jitter
+```
+
+### Fix applicable open issues
+These are the [current](https://github.com/kubernetes/kubernetes/issues/82659) list of issues that are being targeted for GA. 
+
+- [Updating a cronjob causes jobs to be scheduled retroactively](https://github.com/kubernetes/kubernetes/issues/63371)
+- [CLI: Updated CronJob Schedule Missing from Dry Run](https://github.com/kubernetes/kubernetes/issues/73613)
+- [Kubernetes CronJob pods is not getting clean-up when Job is completed](https://github.com/kubernetes/kubernetes/issues/74741)
+- [Infinite ImagePullBackOff CronJob results in resource leak](https://github.com/kubernetes/kubernetes/issues/76570)
+- [Cronjob `spec.schedule` cannot be change when `spec.schedule` value not `"` or `’`](https://github.com/kubernetes/kubernetes/issues/78646)
+- [Kubelet CPU/Memory Usage linearly increases using CronJob](https://github.com/kubernetes/kubernetes/issues/64137)
+- [Stopping cluster overnight prevents scheduled jobs from running after cluster startup](https://github.com/kubernetes/kubernetes/issues/42649)
+
+### Scale Targets for GA
+
+The scale targets for CronJob GA API shall conform to existing [SLIs/SLOs of Kubernetes native types](https://github.com/kubernetes/community/blob/master/sig-scalability/slos/slos.md#kubernetes-slisslos).
+
+The targets are defined by the below suggested maximum limits, which are organized the same way as the [Kubernetes native type thresholds](https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/thresholds.md#kubernetes-thresholds).
+
+#### CronJob Limits
+There should be nothing in the implementation that limits CronJobs per namespace. Overall clusterwide limits of CronJob are important. Cluster wide limits for CronJob should be storage bound since it shares the storage space with all other objects. Determining the appropriate storage limit for a cluster is out-of-scope for this document. For large clusters we propose to support 10000 CronJobs.
+
+#### Frequency of launched jobs
+The number of CronJobs is also sensitive to the API server QPS and the schedule of the individual CronJobs. This translates to the frequency of launched jobs. We could have large number of CronJobs with a spread of schedule that doenst stress the Job API. At the same time we could have a small number of CronJobs that schedule synchronously stressing the Jobs API. The design must be able to easily saturate the API server QPS. The user can setup ratelimits for CronJob and Job APIs using [API Server ratelimting config](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#eventratelimit).
+
+
+
+## V1 API
+Most of the changes are adding additional status fields.
+
+```golang
```

Comment:
Would you make it more obvious what fields are new? Maybe we don't need to include all fields, just include the new ones. If any fields are renamed, call that out too.

Reply:
added a section above calling out the fields added.